eval.ml

```diff
@@ -128,21 +133,99 @@ let get_bool (Lit lit : value) =
      let expected, received = Bool, litteral_type other in
      raise (Type_error { expected; received })
 
-let instruction conf =
-  if conf.pc >= Array.length conf.instrs
-  then Stop
-  else conf.instrs.(conf.pc)
-
 let reduce conf =
   let eval conf e = eval conf.heap conf.env e in
   let resolve instrs label = Instr.resolve instrs label in
   let pc' = conf.pc + 1 in
   assert (conf.status = Running);
-  match instruction conf with
+
+  let instruction =
+    let defaul_exit = (Simple (Lit (Int 0))) in
+    if conf.pc < Array.length conf.instrs
+    then conf.instrs.(conf.pc)
+    else if conf.continuation = []
+    then Stop defaul_exit
+    else assert (false)
+  in
+
+  let build_call_frame formals actuals =
+    let eval_arg env (formal, actual) =
+      match formal with
+      | Instr.ParamConst x ->
+        let value = eval conf actual in
+        Env.add x (Const value) env
+      | Instr.ParamMut x ->
+        let get_var = function[@warning "-4"]
+          | Simple (Var x) -> x
+          | _ -> assert (false)
+        in
+        let get_addr = function
+          | Mut a as adr -> adr
+          | Const _ -> assert (false)
+        in
```

Comment:
I don't like the `assert (false)` in the two functions above because they rely on the check assumption in way that feel fragile to me. That said, I thought about it and I have no obvious fix to propose (I think using a dedicated exception in this case would be nicer). I'll discuss that in a higher-level comment.